server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapper.java

```diff
@@ -281,51 +300,29 @@ private static boolean ruleToOverloaded(RuleDto rule) {
     return rule.getRemediationFunction() != null;
   }
 
+  @CheckForNull
   private static DebtRemediationFunction defaultDebtRemediationFunction(final RuleDto ruleDto) {
     final String function = ruleDto.getDefaultRemediationFunction();
     if (function == null || function.isEmpty()) {
       return null;
     } else {
-      return new DebtRemediationFunction() {
-        @Override
-        public Type type() {
-          return Type.valueOf(function.toUpperCase());
-        }
-
-        @Override
-        public String coefficient() {
-          return ruleDto.getDefaultRemediationCoefficient();
-        }
-
-        @Override
-        public String offset() {
-          return ruleDto.getDefaultRemediationOffset();
-        }
-      };
+      return new DefaultDebtRemediationFunction(
+        DebtRemediationFunction.Type.valueOf(function.toUpperCase()),
+        ruleDto.getDefaultRemediationGapMultiplier(),
+        ruleDto.getDefaultRemediationBaseEffort());
     }
   }
 
+  @CheckForNull
   private static DebtRemediationFunction debtRemediationFunction(final RuleDto ruleDto) {
     final String function = ruleDto.getRemediationFunction();
     if (function == null || function.isEmpty()) {
       return defaultDebtRemediationFunction(ruleDto);
     } else {
-      return new DebtRemediationFunction() {
-        @Override
-        public Type type() {
-          return Type.valueOf(function.toUpperCase());
-        }
-
-        @Override
-        public String coefficient() {
-          return ruleDto.getRemediationCoefficient();
-        }
-
-        @Override
-        public String offset() {
-          return ruleDto.getRemediationOffset();
-        }
-      };
+      return new DefaultDebtRemediationFunction(
+        DebtRemediationFunction.Type.valueOf(function.toUpperCase()),
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define the locale to be used in this String operation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1449)